### ISSUE TITLE
New site - challenge page - forum link

### ISSUE THREAD

**Issue by hokienick** [topcoderinc/cab#31]
If the user hasn't registered for a challenge or the challenge is closed then the "Challenge Discussion"/forum link doesn't show. This makes it difficult to find discussions for closed challenges.

**Comment by hokienick** [comment]
DaraK says:  _Tony said they have to fix the flow there to have the forum link big and clear as when active and it is now added to github for fix: https://github.com/appirio-tech/topcoder-app/issues/914_

**Comment by birdofpreyru** [comment]
@hokienick Tell me more about it, please. As of now, I see the `Challenge Discussion` link for all my past challenges. Displaying this link to the users that had not participated (registered) in a challenge seems to be a bit against the current policies (you have to register and, probably, sign additional NDA to get access to the forum, where more details of the project are disclosed). Also, current TC NDA states that a competitor should destroy everything related to a challenge he has participated within a week since the end of the challenge. Providing access to the forum kind of goes against that rule.

**Comment by Dara-K** [comment]
@birdofpreyru You are right: I brought this up as I wanted to see the forums on a challenge I didn't register. This is useful when we have 2 challenges that are related, like design-design, or wireframe-design to get a better insight into what client wants or see if one idea was already tried by someone else (especially from wf feedback) on a past closed challenge.

Even if it's against the NDA, there are still solutions to find it like the option to search forums.

**Comment by vic-georgiev** [comment]
Interesting topic. I'll try to find the root problem for keeping the challenge discussion forums available: Continuity - preserving important information for consequent challenges/fixes.

I'd like to argue that such important information should be available in another format entirely rather than forums. @Dara-K can you give me an example for such information that you need from a past challenge?

**Comment by Dara-K** [comment]
@vic-topcoder Take challenge: https://www.topcoder.com/challenge-details/30055920/?type=design  which was a refinement for design from this challenge https://www.topcoder.com/challenge-details/30055123/?type=design. We also got some input documents to go through as well. In this case the feedback was hidden even from forums as it was on Invision, but I could read the overall feedback: https://apps.topcoder.com/forums/?module=Thread&threadID=891325&start=0 

On this challenge I had same questions as Kharm did: https://apps.topcoder.com/forums/?module=Thread&threadID=896443&start=0  if we should stick to the 3 axes or not (as you can see he also searched previous challenge). As feedback was hidden he had to ask, otherwise he could read the direct feedback from client which might have answered his question.

From the general feedback I got that they want to see not just numbers, but also how trends go and also that dashboard should tell a story not be just graphs and statistics. This helped me improve my approach to the initial design. 

Hope this helps.

**Comment by vic-georgiev** [comment]
Thanks Daniela! :)

**Comment by Dara-K** [comment]
What's the status of this guys? it's been a while..
Another member brought the issue up as he registered to a challenge and doesn't see the forum link. I registered to same challenge, same issue. 
![pasted_image_at_2017_05_27_09_39_am](https://cloud.githubusercontent.com/assets/25185079/26521224/eaae1a4e-42eb-11e7-948d-8848751bcbb1.png)

**Comment by birdofpreyru** [comment]
@Dara-K I guess, it is a problem in the backend. The link was rendered properly before, and it still renders properly in dev environment (www.topcoder-dev.com). Let me check with Tony.

**Comment by birdofpreyru** [comment]
@Dara-K Actually, while I was writting a message to Tony, it seems it started to work properly and the links are back in place.

**Comment by Dara-K** [comment]
Right, this is fixed - thanks for looking into it.

How about the initial problem? a challenge I haven't registered or is closed still doesn't show the forum links.

**Comment by birdofpreyru** [comment]
As of now, the forum link for a challenge shown to visitors who have any role on that challenge.
